docs/ops/deployment/docker.md

```diff
@@ -24,119 +24,476 @@ under the License.
+
+    chmod 755 custom_entry_point_script.sh
+
+    docker run \
+        --mount type=bind,src=$(pwd),target=/mnt
+        flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %} /mnt/custom_entry_point_script.sh
+    ```
+
+* **extend the Flink image** by writing a custom `Dockerfile` and build a custom image:
+
+    ```dockerfile
+    FROM flink
+    RUN set -ex; apt-get update; apt-get -y install python
+    ADD /host/path/to/flink-conf.yaml /container/local/path/to/custom/conf/flink-conf.yaml
+    ADD /host/path/to/log4j.properties /container/local/path/to/custom/conf/log4j.properties
+    ln -fs /opt/flink/opt/flink-metrics-graphite-*.jar /opt/flink/lib/.
+    ln -fs /opt/flink/opt/flink-s3-fs-hadoop-*.jar /opt/flink/lib/.
+    ENV VAR_NAME value
+    ```
+
+    ```sh
+    docker build -t custom_flink_image .
+    # optional push to your docker image registry if you have it,
+    # e.g. to distribute the custom image to your cluster
+    docker push custom_flink_image
+    ```
+
+{% top %}
+
+## Flink with Docker Compose
+
+[Docker Compose](https://docs.docker.com/compose/) is a convenient way to run a group of Docker containers locally.
+The next chapters show examples of configuration files to run Flink.
 
 ### Usage
 
+* Create the `yaml` files with the container configuration, check examples for:
+    * [Session cluster](#session-cluster-with-docker-compose)
+    * [Single Job](#single-job-cluster-with-docker-compose)
+
+    Check also [the Flink docker image tags](#image-tags) and [how to customize the Flink docker image](#advanced-customization)
+    for usage in the configuration files.
+
 * Launch a cluster in the foreground
 
-        docker-compose up
+    ```sh
+    docker-compose up
+    ```
 
 * Launch a cluster in the background
 
-        docker-compose up -d
+    ```sh
+    docker-compose up -d
+    ```
 
 * Scale the cluster up or down to *N* TaskManagers
 
-        docker-compose scale taskmanager=<N>
+    ```sh
+    docker-compose scale taskmanager=<N>
+    ```
 
-* Kill the cluster
+* Access the Flink Master container
 
-        docker-compose kill
+    ```sh
+    docker exec -it $(docker ps --filter name=jobmanager --format={{.ID}}) /bin/sh
+    ```
 
-When the cluster is running, you can visit the web UI at [http://localhost:8081](http://localhost:8081). 
-You can also use the web UI to submit a job to a session cluster.
+* Kill the cluster
 
-To submit a job to a session cluster via the command line, you must copy the JAR to the JobManager
-container and submit the job from there.
+    ```sh
+    docker-compose kill
+    ```
+
+* Access Web UI
+
+    When the cluster is running, you can visit the web UI at [http://localhost:8081](http://localhost:8081).
+    You can also use the web UI to submit a job to a Session cluster.
+
+* To submit a job to a *Session cluster* via the command line, you must copy the JAR to the Flink Master container and
```

Comment:
maybe also document that you could submit a job like usual if you have a distribution locally; after all if the WebUI is accessible then so is the REST API.

Reply:
is it not written above?

Reply:
ok, now I get it